casper/src/main/scala/coop/rchain/casper/ReportStore.scala

```diff
@@ -0,0 +1,42 @@
+package coop.rchain.casper
+
+import cats.effect.Sync
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.dag.codecs
+import coop.rchain.casper.protocol.BlockEventInfo
+import coop.rchain.shared.Compression
+import coop.rchain.shared.syntax._
+import coop.rchain.store.{KeyValueStoreManager, KeyValueTypedStore}
+import net.jpountz.lz4.{LZ4CompressorWithLength, LZ4DecompressorWithLength}
+import scodec.SizeBound.unknown
+import scodec.{Attempt, Codec, DecodeResult, SizeBound}
+import scodec.bits.BitVector
+
+object ReportStore {
+
+  val compressor = new LZ4CompressorWithLength(Compression.factory.fastCompressor())
+  // val compressor = new LZ4CompressorWithLength(factory.highCompressor(17)) // Max compression
+  val decompressor = new LZ4DecompressorWithLength(Compression.factory.fastDecompressor())
+
+  def compressBytes(bytes: Array[Byte]): Array[Byte] = compressor.compress(bytes)
+
+  type ReportStore[F[_]] = KeyValueTypedStore[F, ByteString, BlockEventInfo]
+
+  val blockEventInfoCodec = new Codec[BlockEventInfo] {
+    override def encode(value: BlockEventInfo): Attempt[BitVector] =
+      Attempt.successful(BitVector(compressor.compress(value.toByteArray)))
+
+    override def decode(bits: BitVector): Attempt[DecodeResult[BlockEventInfo]] =
+      Attempt.successful(
+        DecodeResult(
+          BlockEventInfo.parseFrom(decompressor.decompress(bits.toByteArray)),
+          BitVector.empty
+        )
+      )
+
+    override def sizeBound: SizeBound = unknown
+  }
```

Comment:
There are helper functions to define new Codecs more easily without using Codec constructor directly.
```scala
val blockEventInfoCodecCompressed =
  scodec.codecs.bytes.xmap[BlockEventInfo](
    bv => BlockEventInfo.parseFrom(decompressor.decompress(bv.toArray)),
    bei => ByteVector(compressor.compress(bei.toByteArray))
  )
```

Reply:
I don't like some of the builtin Codec in SCodec because it would append some wired strings in the codec implementation.